### PR TITLE
Improve documentation for the hashes module

### PR DESCRIPTION
**Pull request by konsumlamm**
* use runnableExamples
* add note about ==
* small doc comment changes

### CODE REVIEW

File: lib/pure/hashes.nim [hunk 1]
```diff
@@ -484,8 +503,9 @@ proc hash*[A](x: openArray[A]): Hash =
 proc hash*[A](aBuf: openArray[A], sPos, ePos: int): Hash =
   ## Efficient hashing of portions of arrays and sequences, from starting
   ## position `sPos` to ending position `ePos` (included).
+  ## The elements must have a hash proc.
   ##
-  ## ``hash(myBuf, 0, myBuf.high)`` is equivalent to ``hash(myBuf)``.
+  ## `hash(myBuf, 0, myBuf.high)` is equivalent to `hash(myBuf)`.
```
Comment (dom96) [inline lib/pure/hashes.nim:509]: Why these changes? They make no difference from what I've seen in the docgen.

File: lib/pure/hashes.nim [hunk 2]
```diff
@@ -507,6 +527,7 @@ proc hash*[A](aBuf: openArray[A], sPos, ePos: int): Hash =
 
 proc hash*[A](x: set[A]): Hash =
   ## Efficient hashing of sets.
+  ## The elements must have a hash proc.
```
Comment (dom96) [inline lib/pure/hashes.nim]:
```suggestion
  ## Efficient hashing of sets.
  ##
  ## There must be a ``hash`` proc defined for the type of the elements ``A``.
```

File: lib/pure/hashes.nim [hunk 3]
```diff
@@ -461,14 +479,15 @@ proc hashIgnoreCase*(sBuf: string, sPos, ePos: int): Hash =
...
-  ## Efficient hashing of tuples.
+  ## Efficient hashing of tuples. The fields must have a hash proc.
   for f in fields(x):
     result = result !& hash(f)
   result = !$result
 
 
 proc hash*[A](x: openArray[A]): Hash =
   ## Efficient hashing of arrays and sequences.
+  ## The elements must have a hash proc.
```
Comment (dom96) [inline lib/pure/hashes.nim]:
```suggestion
  ##
  ## There must be a ``hash`` proc defined for the type of the elements ``A``.
```

File: lib/pure/hashes.nim (hunk 1 above)
Comment (konsumlamm) [reply lib/pure/hashes.nim:509]: Ye, it doesn't make a difference in docgen, but single backticks are shorter and I think it looks cleaner (in the source). Also, the contributing guideline says:

> In nim sources, prefer single backticks to double backticks since it's simpler and `nim doc` supports it

File: lib/pure/hashes.nim [hunk 4]
```diff
@@ -10,46 +10,63 @@
...
+## * `std/sha1 module <sha1.html>`_ for a SHA-1 encoder and decoder
 ## * `tables module <tables.html>`_ for hash tables
 
 import std/private/since
 
 type
   Hash* = int ## A hash value. Hash tables using these values should
-               ## always have a size of a power of two and can use the ``and``
-               ## operator instead of ``mod`` for truncation of the hash value.
+              ## always have a size of a power of two and can use the `and`
```
Comment (timotheecour) [inline lib/pure/hashes.nim]:
```suggestion
              ## always have a size of a power of two so they can use the `and`
```

**Review by timotheecour — APPROVED** [review]
LGTM after nits

**Review by timotheecour — APPROVED** [review]
